### PR TITLE
AT-8276-Page2-Security-Requirement-Component

### PR DESCRIPTION
**Pull request by DGabriel-CCPO**
In the event that the Mission Owner selects Secret and/or Top Secret in their classification requirements, the Mission Owner needs to input the security requirements that contractors will need to access. The Mission Owner also needs to input their need for a portability plan if they use Advisory and Assistance Services in Performance Requirements.
AT-8277: SlideOut included

### CODE REVIEW

File: src/steps/04-ContractDetails/SecurityRequirements.vue [hunk 1]
```diff
@@ -1,15 +1,242 @@
...
+              :card="false"
+              class="copy-max-width"
+              :rules="[
+              $validators.required('Please select at least one type of classified information.')
+            ]"
+            />
+          </div>
+          <hr v-if="hasTopSecret && hasSecret" />
+          <div v-if="hasTopSecret">
+            <p id="SecretMessage" class="mb-5 font-weight-500">
```
Comment (ericyoungquist-ccpo) [inline src/steps/04-ContractDetails/SecurityRequirements.vue:50]: update ID to `TopSecretMessage`

File: src/steps/04-ContractDetails/SecurityRequirements.vue [hunk 2]
```diff
@@ -1,15 +1,242 @@
...
+              For your TOP SECRET cloud services and support, what type of classified information
+              will be accessed?
+            </p>
+            <p id="SecretMessage" class="mb-5 ">
```
Comment (ericyoungquist-ccpo) [inline src/steps/04-ContractDetails/SecurityRequirements.vue:54]: update ID to `TopSecretMessageNote`

File: src/steps/04-ContractDetails/SecurityRequirements.vue [hunk 3]
```diff
@@ -1,15 +1,242 @@
...
+              Select all that apply to your TOP SECRET classification level.
+            </p>
+            <ATATCheckboxGroup
+              id="secretSecurityRequirements"
```
Comment (ericyoungquist-ccpo) [inline src/steps/04-ContractDetails/SecurityRequirements.vue:58]: update ID to `TopSecretSecurityRequirements`

File: src/store/classificationRequirements/index.ts [hunk 4]
```diff
@@ -48,6 +55,39 @@ export class ClassificationRequirementsStore extends VuexModule {
       this.sessionProperties,
       ATAT_CLASSIFICATION_LEVELS_KEY
     );
+
+  }
+
+  @Mutation
+  public async setCurrentENVClassificationLevels(value: ClassificationLevelDTO[]): Promise<void> {
```
Comment (ericyoungquist-ccpo) [inline src/store/classificationRequirements/index.ts:62]: pls delete this mutation - it was removed when current env classification requirements was refactored to be in the CurrentEnvironment store

File: src/store/classificationRequirements/index.ts [hunk 5]
```diff
@@ -48,6 +55,39 @@ export class ClassificationRequirementsStore extends VuexModule {
...
+    storeDataToSession(
+      this,
+      this.sessionProperties,
+      ATAT_CLASSIFICATION_LEVELS_KEY
+    );
+
+  }
+
+  @Mutation
+  public async setEnvironmentInstance(value: EnvironmentInstanceDTO[]): Promise<void> {
```
Comment (ericyoungquist-ccpo) [inline src/store/classificationRequirements/index.ts:73]: pls delete this mutation - it was removed when current env classification requirements was refactored to be in the CurrentEnvironment store

**Review by ericyoungquist-ccpo — CHANGES_REQUESTED** [review]

File: src/steps/04-ContractDetails/SecurityRequirements.vue [hunk 6]
```diff
@@ -1,15 +1,242 @@
...
+          <div v-if="hasSecret">
+            <p id="SecretMessage" class="mb-5 font-weight-500">
+              For your SECRET cloud services and support, what type of classified information will
+              be accessed?
+            </p>
+            <p id="SecretMessage" class="mb-5 ">
+              Select all that apply to your SECRET classification level.
+            </p>
+            <ATATCheckboxGroup
+              id="secretSecurityRequirements"
```
Comment (ericyoungquist-ccpo) [inline src/steps/04-ContractDetails/SecurityRequirements.vue:37]: PascalCase this ID pls

File: src/components/DOW/SecurityRequirementsCheckboxes.vue [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div>
+    <v-container fluid class="container-max-width">
+      <v-row>
+        <v-col class="col-12">
```
Comment (ericyoungquist-ccpo) [inline src/components/DOW/SecurityRequirementsCheckboxes.vue:5]: remove `v-container` to `v-col` - adds unwanted padding

File: src/components/DOW/SecurityRequirementsForm.vue [hunk 8]
```diff
@@ -0,0 +1,144 @@
+<template>
+  <div>
+    <v-container fluid class="container-max-width">
+      <v-row>
+        <v-col class="col-12">
```
Comment (ericyoungquist-ccpo) [inline src/components/DOW/SecurityRequirementsForm.vue:5]: remove `v-container` to `v-col` - adds unwanted padding

**Review by ericyoungquist-ccpo — CHANGES_REQUESTED** [review]

**Review by tonysumner-ccpo — APPROVED** [review]

**Review by nkingsley-ccpo — APPROVED** [review]

**Pull request merged**